gc/base/TLHAllocationInterface.cpp

```diff
@@ -309,4 +309,26 @@ MM_TLHAllocationInterface::restartCache(MM_EnvironmentBase *env)
 #endif /* defined(OMR_GC_NON_ZERO_TLH) */
 }
 
+#if defined(OMR_GC_NON_ZERO_TLH)
+uintptr_t
+MM_TLHAllocationInterface::getRemainingCacheSize(bool nonZero)
+{
+	if (nonZero) {
+		return _tlhAllocationSupportNonZero.getRemainingSize();
+	} else {
+		return _tlhAllocationSupport.getRemainingSize();
+	}
+}
+
+uintptr_t
+MM_TLHAllocationInterface::getRefreshCacheSize(bool nonZero)
+{
+	if (nonZero) {
+		return _tlhAllocationSupportNonZero.getRefreshSize();
+	} else {
+		return _tlhAllocationSupport.getRefreshSize();
+	}
+}
+#endif /* defined(OMR_GC_NON_ZERO_TLH) */
```

Comment:
We can go this way if you like. However would it be cleaner if we keep signature of the functions the same for OMR_GC_NON_ZERO_TLH defined or not? Yes, if it is not defined parameter is not going to be used. I don't think compiler is going to complain about this.
If we agree on signatures so check for OMR_GC_NON_ZERO_TLH can be moved to function body, like:
```
uintptr_t
MM_TLHAllocationInterface::getRemainingCacheSize(bool nonZero)
{
#if defined(OMR_GC_NON_ZERO_TLH)
	if (nonZero) {
		return _tlhAllocationSupportNonZero.getRemainingSize();
	} else 
#endif /* defined(OMR_GC_NON_ZERO_TLH) */
    {
		return _tlhAllocationSupport.getRemainingSize();
	}
}
```
Would it be more symmetric and cleaner?

Reply:
It's simpler/cleaner, but not completely semantically correct. Boolean nonZero does not make sense if OMR_GC_NON_ZERO_TLH is not defined. In such a build, why would someone need to bother thinking what to set for that boolean....

I think that OMR_GC_NON_ZERO_TLH build flag was just unnecessary. Dual TLHs should have been allowed to build/compile always (and the existing runtime flag would suffice, since it's of very low runtime overhead), and then your/simpler approach would work.

Ovearall, no strong preferences. Just let me know if you still prefer the simpler approach.

Reply:
I guess OMR_GC_NON_ZERO_TLH was more introduced because of potential footprint concerns in minimalistic builds (size of one more Support struct per thread), rather then runtime overhead (checking if object should be zeroed or not)?

Reply:
I can live with that small semantical anomaly, and simplify it as you suggested...